test_player.py

```diff
@@ -91,3 +91,153 @@ def test_computer_player(self):
             expected_output = "x"
             result = self.player.get_mark()
             self.assertEqual(result, expected_output)
+
+    def test_make_move(self):
+        test_cases = [{'name': "ai should take middle cell on first move",
+                       'board': [[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                       'expected_move': 5},
+                      {'name': "ai should take winning diagonal move",
+                       'board': [["x", "o", "x"], [4, 5, "o"], ["x", "o",  9]],
+                       'expected_move': 4},
+                      {'name': "ai should take winning horizontal move",
+                       'board': [["x", "x", 3], [4, 5, "o"], [7, "o", 9]],
+                       'expected_move': 3},
+                      {'name': "ai should take winning vertical move",
+                       'board': [["x", 2, 3], [4, 5, "o"], ["x", "o", 9]],
+                       'expected_move': 4},
+                      {'name': "ai should block opponent's two-move win",
+                       'board': [['x', 2, 3], [4, 'o', 6], [7, 8, 'x']],
+                       'expected_move': 3},
+                      {'name': "ai should block opponent's diagonal win",
+                       'board': [["o", "x", 3], [4, 5, "x"], [7, 8, "o"]],
+                       'expected_move': 5},
+                      {'name': "ai should block opponent's vertical win",
+                       'board': [["o", "x", 3], ["o", 5, "x"], [7, 8, 9]],
+                       'expected_move': 7},
+                      {'name': "ai should block opponent's horizontal win",
+                       'board': [["x", "o", "x"], ["o", 5, "o"], [7, 8, "x"]],
+                       'expected_move': 5},
+                      {'name': "ai should choose the last move available if all but one move is left",
+                       'board': [["o", "x", "o"], ["x", "o", "x"], ["x", "o", 9]],
+                       'expected_move': 9}
```

Comment:
Now that you've got a solid understanding of test structure and naming, it really shows in these tests! 
I really like how you are explaining what behavior you are checking and helps with finding the edge cases.
Well done :)